File: 0997-find-the-town-judge/0997-find-the-town-judge.py
class Solution:
    def findJudge(self, n: int, trust: List[List[int]]) -> int:
        
        if n==1: return 1
        follower = defaultdict(int)
        followed = defaultdict(int)

        for ix in range(len(trust)):
            follower[trust[ix][0]] +=1 
            followed[trust[ix][1]] +=1 
        for ix in range(1,n+1):
            if followed[ix] == n-1:
                if follower[ix] ==0: 
                    return ix 
        return  -1 

File: 0997-find-the-town-judge/0997-find-the-town-judge.java
class Solution {
    public int findJudge(int n, int[][] trust) {
        
        int[] follower = new int[n+1] ;
        int[] followed = new int[n+1] ;

        for(int[] t:trust){
            follower[t[1]]++ ; 
            followed[t[0]]++ ;
        }
        for(int ix=1 ; ix <=n ; ix++){
            if( follower[ix] ==n-1 && followed[ix] ==0 ){
                return ix ; 
            }
        }
        
        return -1  ; 

    }
}